### PR TITLE
Outputs scripts for FSDP runs

### PR DESCRIPTION
**Pull request by flohump**
## :bird: Purpose of this PR :bird:

- The central script for collecting FSDP outputs at regional, iso and grid level is `FSDP_collect.R`
- As soon as other gridded output become available, they should be add to `FSDP_collect.R` in the same way as for BII and land.
- The script also creates plots. For this we need the `[m4fsdp]` R library on the cluster. https://github.com/pik-piam/m4fsdp
- No test runs needed. Just output scripts.

## :wrench: Checklist for PR creator :wrench:

- [x] Labeling pull request correctly [from the label list](https://github.com/magpiemodel/magpie/labels).

* Low risk : Simple bugfixes (missing files, updated documentation, typos) or Start/output scripts
* Medium risk : New realization / Changes to existing realization / Other changes which don't modify default.cfg
* High risk : New input files (if cfg$input is changed in default.cfg) / Modification to core model (eg. changes in equations, calculations, introduction of new sets etc.) / Other changes in default.cfg

- [x] Providing additional information based on PR label

* Low risk : No new model run needed.
* Medium risk : Default run based on the current version of the fork from which PR is made
* High risk
  * Default run from the current develop branch
  * Default run based on the current version of the fork from which PR is made

- [x] :chart_with_downwards_trend: Performance loss/gain from current default behavior :chart_with_upwards_trend:
  * Current develop branch default : ** mins
  * This PR's default :  ** mins

- [x] Added changes to `CHANGELOG.md`
- [x] Compilation check (model starts without compilation errors - use `gams main.gms action=c` in model folder for testing).
- [x] No hard coded numbers and cluster/country/region names.
- [x] The new code doesn't contain declared but unused parameters or variables.
- [x] Where relevant, In-code comments added including documentation comments.
- [x] Made sure that documentation created with [`goxygen`](https://github.com/pik-piam/goxygen) is okay (use `goxygen::goxygen()` for testing).
- [x] Changes to [`magpie4`](https://github.com/pik-piam/magpie4) R library for post processing of model output (ideally backward compatible).
- [x] Self-review of my own code.
- [x] For high risk runs: validation of major model indicators - Land-use, emissions, food prices, Tau.  %Delete this line in case it is not a high risk run%

## :warning: Additional notes or warnings :warning:
NA

## :rotating_light: Checklist for RSE reviewer :rotating_light:

- [ ] PR is labeled correctly.
- [ ] `CHANGELOG` is updated correctly
- [ ] No hard coded numbers and cluster/country/region names.
- [ ] No unnecessary increase in module interfaces
- [ ] All required updates in interfaces (if any) have been properly adressed in the module contracts
- [ ] In-code comments and documentation comments are satisfactory.
- [ ] model behavior/performance is satisfactory.
- [ ] Requested changes (if any) were applied correctly

## :rotating_light: Checklist for MAgPIE reviewer :rotating_light:

- [ ] PR is labeled correctly.
- [ ] `CHANGELOG` is updated correctly
- [ ] No hard coded numbers and cluster/country/region names.
- [ ] Changes to the model are scientifically sound
- [ ] In-code comments and documentation comments are satisfactory.
- [ ] model behavior/performance is satisfactory.
- [ ] Requested changes (if any) were applied correctly

### CODE REVIEW

File: scripts/output/rds_report_iso.R [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+library(magclass)
+library(magpie4)
+library(lucode2)
+library(quitte)
+library(gms)
+options("magclass.verbosity" = 1)
+
+############################# BASIC CONFIGURATION #############################
+if(!exists("source_include")) {
+  outputdir <- "/p/projects/landuse/users/miodrag/projects/tests/flexreg/output/H12_setup1_2016-11-23_12.38.56/"
```
Comment (k4rst3ns) [inline scripts/output/rds_report_iso.R:24]: These seems strange. For the other script you had something more generic and not hard coded.

**Review by k4rst3ns — COMMENTED** [review]
Looks good to me.

Two comments:
1. some of the filter statements (like `x <- unlist(lapply(strsplit(basename(outputdir),"_"),function(x) x[2]))` or `if(unlist(strsplit(title,"_"))[3] %in% scen)`) seem to work only with a specific naming convention and are not to easy to grasp. I guess this works fine for now, for long term and more generic applicability it might be cool to find more generic statements maybe.
2. Maybe let a lintr run over the scripts to improve coding style (like spaces, line length etc.) for better readibility.

File: scripts/output/projects/FSDP_collect.R [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+# Version 1.0, Florian Humpenoeder
+#
+library(lucode2)
+library(magclass)
+library(gms)
+library(magpiesets)
+library(data.table)
+library(gdx)
+options(error=function()traceback(2))
```
[the remark this answers is not in the document]
Comment (tscheypidi) [reply scripts/output/projects/FSDP_collect.R]: please remove as these options might also affect follow up calculations in an unanticipated way

**Review by tscheypidi — APPROVED** [review]

File: scripts/output/projects/FSDP_collect.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (flohump) [reply scripts/output/projects/FSDP_collect.R]: re 1) The output script is very specific to the FSDP runs. In included a version flag for future FSDP runs. With this version flag some the strsplit statement can be removed in the future.

**Review by k4rst3ns — APPROVED** [review]
All good for me now.

**Review by bodirsky — APPROVED** [review]

**Comment by bodirsky** [comment]
Mergin this as Florian is not here anymore!

**Pull request merged**